src/build/env.py

```diff
@@ -182,9 +182,27 @@ def _create_isolated_env_venv(path):  # type: (str) -> Tuple[str, str]
     pip_distribution = next(iter(metadata.distributions(name='pip', path=[purelib])))
     pip_version = packaging.version.Version(pip_distribution.version)
 
-    # Currently upgrade if Pip 19.1+ not available, since Pip 19 is the first
-    # one to officially support PEP 517, and 19.1 supports manylinux1.
+    needs_pip_upgrade = False
     if pip_version < packaging.version.Version('19.1'):
+        # PEP-517 and manylinux1 was first implemented in 19.1
+        needs_pip_upgrade = True
+    elif platform.system() == 'Darwin':
+        # macOS 11+ name scheme change requires 20.3. Intel macOS 11.0 can be told to report 10.16 for backwards
+        # compatibility; but that also fixes earlier versions of pip so this is only needed for 11+.
+
+        #   TODO: remove when upgrading mypy from 0.800 or dropping Python 2
+        if sys.version_info >= (3,):
+            platform_mac_ver = platform.mac_ver
+        else:
+            platform_mac_ver = cast(Any, platform.mac_ver)
+
+        if int(platform_mac_ver()[0].split('.')[0]) >= 11:
```

Comment:
Could we save the string in `platform_mac_ver` instead of the callable? I fell that is a bit clearer.